api/envoy/type/format.proto

```diff
@@ -0,0 +1,44 @@
+syntax = "proto3";
+
+package envoy.type;
+
+option java_outer_classname = "FormatProto";
+option java_multiple_files = true;
+option java_package = "io.envoyproxy.envoy.type";
+
+import "google/protobuf/struct.proto";
+
+// [#protodoc-title: Format]
+
+// Allows to define custom format of returned data.
+message StringOrJson {
+
+  oneof format {
+    // Plain text format. Allows to specify some custom message to returns.
+    //
+    // .. code-block:: yaml
+    //
+    //   string_format: "My custom message"
+    //
+    string string_format = 1;
+
+    // Json format as dictionary. Allows to specify some custom message to return in json format.
+    //
+    // .. code-block:: yaml
+    //
+    //  json_format:
+    //    protocol: "HTTP/1.1"
+    //    message: "My message"
+    //
+    // The following JSON object would be returned:
```

Comment:
Can you rephrase the comments in this file to make it generic, i.e. it might be used in request, response, aribtrary config, etc. It's just a way of specifying some data that might be either structured JSON or flat plain text.